### PR TITLE
Adding support for LayoutLMvX variants for `object-detection`.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Adding support for `layoutlm` to `object-detection`.

LayoutLMv{2,3} can be used for `object-detection`.

However the classes are `ForTokenClassification` which not all classes can support
a vision + OCR type of inference. (the model needs `bbox` object even if we splitted
out the OCR).

The current implementation changes `object-detection` to `multimodal` since
now the models require `tokenizer` for the layoutlm variants. (This does not 
affect existing working pipelines).

Then it uses reflection at runtime to see if the model is using a `tokenizer`.
This is not a great way to go about it but was the "simpler" change I could think
of.
As long as we don't have support for other model architecture, I'm hesitant to
make "cleaner" modifications, since I don't know if other architectures will
support the same invariants or not.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/pipelines/object_detection.py [hunk 1]
```diff
@@ -82,33 +84,60 @@ def preprocess(self, image):
...
-        raw_annotation["labels"] = [self.model.config.id2label[label.item()] for label in labels]
-        raw_annotation["boxes"] = [self._get_bounding_box(box) for box in boxes]
-
-        # {"scores": [...], ...} --> [{"score":x, ...}, ...]
-        keys = ["score", "label", "box"]
-        annotation = [
-            dict(zip(keys, vals))
-            for vals in zip(raw_annotation["scores"], raw_annotation["labels"], raw_annotation["boxes"])
-        ]
+        if self.tokenizer:
```
Comment (sgugger) [inline src/transformers/pipelines/object_detection.py:102]:
```suggestion
        if self.tokenizer is not None:
```

**Review by sgugger — APPROVED** [review]
Thanks for adding this! Just have a couple of nits.

File: src/transformers/pipelines/object_detection.py (hunk 1 above)
Comment (Narsil) [reply src/transformers/pipelines/object_detection.py]: Done.

File: src/transformers/pipelines/object_detection.py [hunk 2]
```diff
@@ -82,33 +84,60 @@ def preprocess(self, image):
...
-        raw_annotation["boxes"] = [self._get_bounding_box(box) for box in boxes]
-
-        # {"scores": [...], ...} --> [{"score":x, ...}, ...]
-        keys = ["score", "label", "box"]
-        annotation = [
-            dict(zip(keys, vals))
-            for vals in zip(raw_annotation["scores"], raw_annotation["labels"], raw_annotation["boxes"])
-        ]
+        if self.tokenizer is not None:
+            w, h = target_size[0].tolist()
```
Comment (sgugger) [inline src/transformers/pipelines/object_detection.py:103]: Full names here please :-)

Comment (sgugger) [reply src/transformers/pipelines/object_detection.py:103]: (Not sure why it didn't go through in my review)

File: src/transformers/pipelines/object_detection.py [hunk 3]
```diff
@@ -82,33 +84,60 @@ def preprocess(self, image):
...
-        raw_annotation["labels"] = [self.model.config.id2label[label.item()] for label in labels]
-        raw_annotation["boxes"] = [self._get_bounding_box(box) for box in boxes]
-
-        # {"scores": [...], ...} --> [{"score":x, ...}, ...]
-        keys = ["score", "label", "box"]
-        annotation = [
-            dict(zip(keys, vals))
-            for vals in zip(raw_annotation["scores"], raw_annotation["labels"], raw_annotation["boxes"])
-        ]
+        if self.tokenizer is not None:
```
Comment (alaradirik) [inline src/transformers/pipelines/object_detection.py:102]: Could you add a comment above the if else statement to clarify which model this is for (for future use purposes)?

**Review by alaradirik — APPROVED** [review]
Thank you for adding this, looks good to me!

File: src/transformers/pipelines/object_detection.py (hunk 2 above)
Comment (Narsil) [reply src/transformers/pipelines/object_detection.py:103]: :)

File: src/transformers/pipelines/object_detection.py (hunk 3 above)
Comment (Narsil) [reply src/transformers/pipelines/object_detection.py:102]: Done.

**Pull request merged**